In [1]:
import pandas as pd

In [2]:
t17 = pd.read_csv("data/raw_tide2017.csv")
t17.head(24)

Date Time (GMT)  Predicted (ft) Preliminary (ft)  Verified (ft)
0   2017/01/01      00:00          -0.103                -           0.31
1   2017/01/01      01:00          -0.556                -          -0.19
2   2017/01/01      02:00          -0.467                -          -0.09
3   2017/01/01      03:00           0.108                -           0.62
4   2017/01/01      04:00           1.029                -           1.44
5   2017/01/01      05:00           2.062                -           2.63
6   2017/01/01      06:00           2.950                -           3.38
7   2017/01/01      07:00           3.530                -           4.03
8   2017/01/01      08:00           3.733                -           4.32
9   2017/01/01      09:00           3.567                -           3.83
10  2017/01/01      10:00           3.172                -           3.49
11  2017/01/01      11:00           2.750                -           3.05
12  2017/01/01      12:00           2.469                -           2.68
13  2017/01/01      13:00           2.463                -           2.84
14  2017/01/01      14:00           2.790                -           3.18
15  2017/01/01      15:00           3.391                -           3.82
16  2017/01/01      16:00           4.136                -           4.42
17  2017/01/01      17:00           4.837                -           5.13
18  2017/01/01      18:00           5.273                -           5.66
19  2017/01/01      19:00           5.295                -           5.71
20  2017/01/01      20:00           4.863                -           4.86
21  2017/01/01      21:00           3.997                -           4.30
22  2017/01/01      22:00           2.834                -           3.33
23  2017/01/01      23:00           1.610                -           1.85

Above data is great, will get rid of predicted and preliminary columns. Challenge is to figure out how to merge all the data I have together based on time stamp, since time is prob recorded differently in each dataset anddd GMT is useless for this project, gotta convert to PST eventually. The cdip data might actually also be on GMT because it had the time listed as UTC, and UTC +0 = GMT. Will need to switch to PST so that I can eventually drop all of the observations that occur at night.

In [3]:
t17.shape

(8760, 5)

Gotta check shape of each tide file to make sure they have the correct amount of data - 8760 entries, one for each hour, and 8784 entries in 2020 cuz leap year

In [4]:
t18 = pd.read_csv("data/raw_tide2018.csv")
t19 = pd.read_csv("data/raw_tide2019.csv")
t20 = pd.read_csv("data/raw_tide2020.csv")
t21 = pd.read_csv("data/raw_tide2021.csv")

In [5]:
t18.shape

(8760, 5)

In [6]:
t19.shape

(8760, 5)

In [7]:
t20.shape

(8784, 5)

In [8]:
t21.shape

(8760, 5)

Yew all looks good

# Tide has perfect dimensions, it's UTC should be base to merge on

In [10]:
t21['Date'][0]

'2021/01/01'

Gotta replace \slashes with -dashes

In [13]:
t21['Date'] = t21['Date'].str.replace('/','-')

t21['UTC'] = t21['Date']+' '+t21['Time (GMT)']

t21['UTC'] = pd.to_datetime(t21['UTC'], utc=True)

t21['Tide'] = t21['Verified (ft)']

In [28]:
t21.head()

Date Time (GMT)  Predicted (ft) Preliminary (ft)  Verified (ft)  \
0  2021-01-01      00:00          -0.665                -          -0.56   
1  2021-01-01      01:00          -0.943                -          -0.94   
2  2021-01-01      02:00          -0.672                -          -0.43   
3  2021-01-01      03:00           0.057                -           0.34   
4  2021-01-01      04:00           1.067                -           1.26   

                        UTC                      UTC1  Tide  
0 2021-01-01 00:00:00+00:00 2021-01-01 00:00:00+00:00 -0.56  
1 2021-01-01 01:00:00+00:00 2021-01-01 01:00:00+00:00 -0.94  
2 2021-01-01 02:00:00+00:00 2021-01-01 02:00:00+00:00 -0.43  
3 2021-01-01 03:00:00+00:00 2021-01-01 03:00:00+00:00  0.34  
4 2021-01-01 04:00:00+00:00 2021-01-01 04:00:00+00:00  1.26

In [29]:
to_drop = ['Date', 'Time (GMT)', 'Predicted (ft)', 'Preliminary (ft)',
       'Verified (ft)']

t21.drop(columns = to_drop).head()

UTC                      UTC1  Tide
0 2021-01-01 00:00:00+00:00 2021-01-01 00:00:00+00:00 -0.56
1 2021-01-01 01:00:00+00:00 2021-01-01 01:00:00+00:00 -0.94
2 2021-01-01 02:00:00+00:00 2021-01-01 02:00:00+00:00 -0.43
3 2021-01-01 03:00:00+00:00 2021-01-01 03:00:00+00:00  0.34
4 2021-01-01 04:00:00+00:00 2021-01-01 04:00:00+00:00  1.26

In [30]:
t21['UTC'].dtype

datetime64[ns, UTC]

In [33]:
t21['UTC'][0].tzinfo.utcoffset(None)

datetime.timedelta(0)